### DISCUSSION TITLE
Update with a tag-list

### DISCUSSION THREAD

**Discussion by zerswager** [trickerer01/Ruxx#18] [Ideas]
I frequently use your software and often need to update a large number of tags due to new content being added daily. Would it be possible to add a feature that allows the software to update tags using a list?

**Comment by trickerer01** [comment]
Just provide all the necessary details and I'll add this feature for you.
- Which modules will be affected
- What exactly do you want updated and where
- Structure of the input (show me what kind of list you want to use, may just attach its shortened version)

**Reply by zerswager** [reply]
The It RX modules have worked best for me so far. would be really helpful to add a new function to the navbar in the software, perhaps called "Tag List." This would make things much easier and more organized! With this function, you wouldn’t need to rerun the same tags every week just to update your folders.

The function could be a notepad where you list all the tags you want to update regularly. For example:

- yoimiya_(genshin_impact)
- lumine_(genshin_impact) +bunny_girl
- yae_miko +bunny_girl

This approach could save a lot of time and keep everything neat and efficient.

**Reply by trickerer01** [reply]
I have to clarify things here because your first post translation was completely misleading and your request is not about tags but a download feature.

So you want to use simple tags list as input. Ok. And Ruxx has to do what? Do you want the app to simply process all the tags in the list OR you want it to work in background and periodically check for new posts using these tags? I mean with either approach you have to run the app at least once a week, although it indeed would save time (depending on tags count).
Both of these features I did implement in my video downloaders before. You can read about the first one here (as an example): https://github.com/trickerer01/RV, search for `--download-scenario`, description for the second one is unfortunately only available in help message under `-lookahead` argument help (`python ids.py --help`).
Let me know which one is the closest to what you wanted.

**Reply by zerswager** [reply]
This is exactly what I want – the app should simply process all the tags in the list. It doesn't need to be extra and can run in the background to check for new content available for download.

**Reply by trickerer01** [reply]
Alright. If I implement this feature exactly as a download scenario it will still require tags to search for, so maybe I'll have to do something else.

Meanwhile you may want to take a look at downloader wrapper project I have created some time ago for periodic downloads, supporting any number of queries, id range autoupdate, logs and many other features, although it uses its own syntax for query building: https://github.com/trickerer01/download-multi-async-wrapper. I made it to process multiple downloaders/modules at once. See `SCRIPTING_SYNTAX.txt` if you're insterested.

**Reply by zerswager** [reply]
I will look into that

**Reply by trickerer** [reply]
https://github.com/trickerer01/Ruxx/commit/0e02452026d0315032fdbf08e5d9699a63c871a0

This is a GUI only feature. Check [README](https://github.com/trickerer01/Ruxx?tab=readme-ov-file#misc--tools) on this new feature (`Actions -> Batch download using tag list...`).

Please mark this discussion as answered after you test it.

P.S. I guess you could always just create a batch file to process multiple cmdlines [from console](https://github.com/trickerer01/Ruxx?tab=readme-ov-file#using-from-console) - yes you can use pre-built executable as a cmdline tool too - but it doesn't hurt to have a GUI tool for that.

**Reply by zerswager** [reply]
sorry for late reply I had family issues lately. I will look into that

**Comment by justanalt-nsfw** [comment]
Hi, sorry to hop-in on an old discussion.
Would it be possible to add an option for when tag lists are used the results can be split into respective folders for each tag?

I.e. taglist has:
`tag1
tag2
tag3`

And the output would be 3 folders each with the respective search and download results?

**Reply by trickerer01** [reply]
Discussions wasn't a proper place for a feature request in the first place.

I'll see what I can do but next time open a new issue please.

**Reply by trickerer01** [reply]
https://github.com/trickerer01/Ruxx/commit/84602e88c61a3bced9b96211c6f47b5b2afc3a24

Keep in mind that this may intoduce files duplication in case search results intersect with one another (use deduplication tool if needed). Subfolder name will contain task number (auto zero padded) but its max name length is non-configurable (see `app_defines.py` if you want to change it)

**Reply by justanalt-nsfw** [reply]
Awesome, just tested it.
Thank you very much
